kotlin/workflow-runtime/src/main/java/com/squareup/workflow/LaunchWorkflow.kt

```diff
@@ -140,11 +140,11 @@ fun <InputT, OutputT : Any, RenderingT, ResultT> launchSingleOutputWorkflowIn(
     output: Deferred<OutputT>
   ) -> ResultT
 ): ResultT = launchWorkflowIn(scope, workflow, inputs, initialSnapshot) { renderings, outputs ->
-  val workflowScope = this
+//  val workflowScope = this
```

Comment:
@zach-klippenstein We're nearly there, but there's a teardown race or something going on. If I don't comment out this block, the new `output.await()` call in `WorfklowRunnerViewModel` never fires, and so the call to `finish()` in `HelloWorkflowActivity` is never made, and the app hangs.

Reply:
Maybe we should just implement this behavior in the view model.

Reply:
Yeah, that works nicely. I think we'll want to revert the `launchWorkflowIn` PR, just lost our only use case.

Reply:
Did I miss a test case?

Reply:
Ok, this doesn't actually surprise me. The cancellation does feel pretty racy. It's easy enough to do this manually that providing an API just to do it is redundant and over-engineered anyway.

Reply:
https://github.com/square/workflow/pull/462